content/blog/2015-06-03-service-discovery-introduction.md

```diff
@@ -0,0 +1,328 @@
+can be helpful.
+
+
+## Prometheus and targets
+
+For proper understanding of this blog post, we have to take a look at how Prometheus sees targets.
+To Prometheus each scrape target is nothing more but another set of labels – even its address, which
+is defined by the `__address__` label.
+
+In configurations we have a hierarchie of labels. There are global labels, which are assigned to
+every target scraped by the Prometheus instance. One level above is the `job` label, that is defined directly
+in the configuration file. On the next level are labels defined in the target group from which
+a label is derived. Finally, there are labels that are set per-target as we will see soon.
+
+Each layer of labels overwrites any colliding labels from the layers below. Eventually, we have a flat
+set of labels that describe a single target. Those labels are then attached to every metric that
+is scraped from this target.
+
+
+## Scrape configurations and relabeling
+
+Aside from switching from the ASCII protocol buffer format to YAML, a fundamental change to 
+Prometheus's configuration is the switch from per-job to scrape configurations.
+For simple setups the two can be thought of as almost equivalent.
+At the same time, scrape configurations allow greater flexibility for more advanced use cases.
+
+Each scrape configuration is assigned a job name which serves as a default value for the
+`job` label. The `job` label can then be redefined for whole target groups or single targets.
+For example, we can define two target groups, each of which defines targets for one job. 
+To scrape them with the same parameters, we can configure them as follows:
+
+```
+scrape_configs:
+- job_name: 'overwritten-default'
+
+  scrape_interval: 10s
+  scrape_timeout:  5s
+
+  target_groups:
+  - targets: ['10.1.200.130:5051', '10.1.200.134:5051']
+    labels:
+      job: 'job1'
+
+  - targets: ['10.1.200.130:6220', '10.1.200.134:6221']
+    labels:
+      job: 'job2'
+```
+
+Through a mechanism named _relabeling_ any label can be removed, created, or modified on a per-target level. 
+This allows fine-grained labeling that can also enable meta-data coming from the service discovery.
+Relabeling is the highest layer of label-assignment and overwrites any labels previously set.
+
+Relabeling works as follows:
+
+- A list of source labels is defined
+- For each target the values of those labels are concatenated with a separator
+- A regular expression is matched against the resulting string
+- A new value based on those matches is assigned to another label
+
+Mutiple relabeling rules can be defined for each scrape configuration. A simple one
+that squashes two labels into one, looks as follows:
+
+```
+relabel_configs:
+- source_labels: ['label_a', 'label_b']
+  separator:     ';'
+  regex:         '(.*);(.*)'
+  replacement:   '${1}-${2}'
+  target_label:  'label_c'
+```
+
+This rule transforms a target with the label set
```

Comment:
colon at the end, and then start the following sentence with `...` (and also end it with a colon)?